### DISCUSSION TITLE
Feature Request write # for MP3's

### DISCUSSION THREAD

**Discussion by DrGuns4Hands** [openaudible/openaudible#549] [Ideas]
An extremely useful feature would be if the MP3 conversion also filled out the # field. This is the Book enumeration within the series  1, 2, 2.5, etc. This is one of the few things that broke my workflow and looks like it'll take some manual intervention on my part.

I downloaded 955 books. Used some Excel scripting to massage the data, used an Excel add-on Kutools to create auto generate the folder/subfolder structure based on the massaged data, then used an Excel sheet with some VBA code to transfer all the mp3's into the correct subfolders, and at the end I'll still need to go through and manually assign a # to each book in the series. All things considered the fact that I was able to do the rest of it in about an hour or 2 means I'm still way better off than people that went down the manual route, but you would make my life so much easier if we could tuck that feature in. Not sure if the API you are using gets you that series sequence number though.

P.S. Booksonic uses these numbers for ordering the books in the same folder without having to rename them with numbers at the front, which prevents me from breaking alphebetical views.

P.P.S. Thanks for this product. I now have a backup of my whole collection so if Audible were to go belly up for any reason in the future I don't lose the value of a well equipped car.

P.P.P.S. turns out you can't use decimals in the enumeration, not the most common thing, but I guess I'd need to shift those manually even if they were auto added and dropped the decimal

**Comment by openaudible** [comment]
We now put the 1/1 for the track number in converted files.
We are working on a custom file name system where you can specify the file name based on any meta data info-- author,genre,series,etc. Will probably go in beta some time in Oct 2021.
